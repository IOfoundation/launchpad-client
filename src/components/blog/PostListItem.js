import React from 'react';
import {PropTypes} from 'prop-types';

const PostListItem = props => {
  const {title, description, tag, date} = props;

  return (
    <div className="post-lists-item">
      <h3>{title}</h3>
      <p>{description}</p>
      <div className="post-lists-item__tags">
        <span className="post-lists-item__tags__name">{tag}</span>
        {' | '}
        <span className="post-lists-item__tags__date">{date}</span>
      </div>
    </div>
  );
};

PostListItem.propTypes = {
  date: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  tag: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default PostListItem;
