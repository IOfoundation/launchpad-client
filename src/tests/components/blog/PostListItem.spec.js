import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import PostListItem from '../../../components/blog/PostListItem';

const props = {
  date: '05/09/2019',
  description: 'testing description',
  tag: 'tag',
  title: 'title',
};

describe('<PostListItem />', () => {
  it('renders snapshot of Category', () => {
    const wrapper = shallow(<PostListItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
