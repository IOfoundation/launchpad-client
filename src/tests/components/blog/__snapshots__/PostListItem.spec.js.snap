// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostListItem /> renders snapshot of Category 1`] = `
<div
  className="post-lists-item"
>
  <h3
    className="post-lists-item__title text-bold"
  >
    title
  </h3>
  <p
    className="post-lists-item__description"
  >
    testing description
  </p>
  <div
    className="post-lists-item__tags"
  >
    <span
      className="post-lists-item__tags__name text-semi"
    >
      tag
    </span>
     | 
    <span
      className="post-lists-item__tags__date"
    >
      05/09/2019
    </span>
  </div>
</div>
`;
